clientconn.go

```diff
@@ -75,8 +75,6 @@ var (
 	// errConnClosing indicates that the connection is closing.
 	errConnClosing = errors.New("grpc: the connection is closing")
 	errNoAddr      = errors.New("grpc: there is no address available to dial")
-	// minimum time to give a connection to complete
-	minConnectTimeout = 20 * time.Second
```

Comment:
This timeout was not actually being used anywhere. If you look at the timeout passed to transport.NewClientTransport, you'll see that it's only non-zero if it is passed in ConnectOptions, which is never the case.

If you want to use `max(backoff,minConnectTimeout)`, that would be a change of behaviour, but I can do it.